common/flatpak-utils.c

```diff
@@ -4210,23 +4210,38 @@ static gboolean
 copy_icon (const char        *id,
            GFile             *icons_dir,
            OstreeRepo        *repo,
+           GFile             *root,
            OstreeMutableTree *size_mtree,
            const char        *size,
            GError           **error)
 {
   g_autofree char *icon_name = g_strconcat (id, ".png", NULL);
   g_autoptr(GFile) size_dir = g_file_get_child (icons_dir, size);
   g_autoptr(GFile) icon_file = g_file_get_child (size_dir, icon_name);
+  g_autoptr(GFileInfo) file_info = NULL;
+  g_autoptr(GFile) f = NULL;
   const char *checksum;
+  const char *target_icon_name = icon_name;
 
   if (!ostree_repo_file_ensure_resolved (OSTREE_REPO_FILE(icon_file), NULL))
     {
       g_debug ("No icon at size %s for %s", size, id);
       return TRUE;
     }
 
+  f = g_file_resolve_relative_path (root, g_file_get_path(icon_file) );
+  file_info = g_file_query_info (f, OSTREE_GIO_FAST_QUERYINFO,
+                                 G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
+                                 NULL, error);
+
+  if (g_file_info_get_file_type (file_info) == G_FILE_TYPE_SYMBOLIC_LINK)
+  {
+      target_icon_name = g_file_info_get_attribute_byte_string (file_info, "standard::symlink-target");
+      icon_file = g_file_get_child (size_dir, target_icon_name);
```

Comment:
This however, need to be g_file_resolve_relative_path() to handle symlinks that are not just a filename.